k8s/upgrade/src/common/error.rs

```diff
@@ -677,7 +677,7 @@ pub enum Error {
     },
 
     #[snafu(display("failed to list CustomResourceDefinitions: {source}"))]
-    ListCrds { source: kube::Error },
+    ListCrds { source: Box<Error> },
```

Comment:
Do these still need to be box?

Reply:
The actual error type will depend on the Enum variant.

Reply:
Oh this Error is Self here? So you want to stack up these errors?
What's the use case for this?
If we really need this, I would say do this for the ones which really require this and not throughout. We had this kind of thing in the data-plane and this becomes very complicated to match against.

Reply:
I've removed the `Box<Error>`s. The underlying `kube::Error` should be able to describe the issue, may be unnecessary to wrap it with explicit Snafu variants.